code/modules/underworld/underworld.dm

```diff
@@ -39,6 +39,130 @@
 	bound_height = 96
 	anchored = TRUE
 	density = TRUE
+	max_integrity = 9999
+
+/obj/structure/underworld/carriage_normal/New(loc, ...)
+	. = ..()
+	var/pos = get_step(loc, EAST)
+	pos = get_step(pos, EAST)
+	new /obj/structure/carriagedoor(pos)
+
+#define CANT_LEAVE_ITEMS \
+	/obj/item/rogueweapon/woodstaff/aries, \
+	/obj/item/clothing/head/roguetown/crown, \
+	/obj/item/key/lord
+
+/obj/structure/carriagedoor
+	name = "Carriage door"
+	desc = "The wooden entrance to the carriage. It allows those who come to arrive and those who leave to depart."
+	icon = 'icons/roguetown/misc/doors.dmi'
+	icon_state = "carriage_door"
+	layer = WALL_OBJ_LAYER
+	plane = GAME_PLANE_UPPER
+	pixel_y = 27
+	max_integrity = 9999 // In case any maniac decides to hit it.
+
+
+/obj/structure/carriagedoor/attack_hand(mob/user)
+	. = ..()
+
+	var/mob/living/carbon/human/H
+	if(ishuman(user))
+		H = user
+
+	if(!isliving(user) || user.incapacitated())
+		return //No ghosts or incapacitated folk allowed to do this.
+
+	if(user.get_active_held_item())
+		to_chat(user, span_warn("I cant turn the handle. My hand has something in it!"))
```

Comment:
```suggestion
		to_chat(user, span_warning("I can't turn the handle. My hand has something in it!"))
```
Also, is it even possible to use `attack_hand` with an item in your active hand? This should probably just be removed.